### DISCUSSION TITLE
Difference in word coordinate information

### DISCUSSION THREAD

**Discussion by yavuzKomecoglu** [jsvine/pdfplumber#552] [Ask for help with specific PDFs]
Hello, in some Turkish newspaper PDFs, the word coordinates are below a certain pixel. For example, while the headlines in the news in [2_eylul_15_11_2018.pdf / page-3](https://drive.google.com/file/d/1sOyUXKJLvWDBGkV172i4gx-amjJD94sC/view?usp=sharing) and [2_eylul_30_11_2018.pdf / page-7](https://drive.google.com/file/d/1E75I3EI26uNAu06N2n1M63IhOIUMR9zX/view?usp=sharing) start from the bottom, the coordinates are obtained correctly in the news in [gaziantep_dogus_10_08_2021.pdf / page-4](https://drive.google.com/file/d/1Bk27poBvbLDA4EeGbGTnSi4a1_phu1N8/view?usp=sharing).
What exactly is the difference between them? Why does this occur? What's the difference with these pdf's that such coordinate information returns differently?
Thanks.

First, the words are extracted.
```
data_words = crop.dedupe_chars().extract_words(x_tolerance=1, y_tolerance=1, extra_attrs=["fontname", "size"])           
```

Then the title region is determined.
```
h_left_x0_all = min([h["x0"] for h in title])
h_left_top_all = min([h["top"] for h in title])
h_right_x1_all = max([h["x1"] for h in title])
h_right_bottom_all = max([h["bottom"] for h in title])

title_area = (h_left_x0_all, h_left_top_all), (h_right_x1_all, h_right_bottom_all)
```
Note: Where title_area is absolute position, relative position is used when drawing title area with opencv.

Test Newspapers
- [2_eylul_15_11_2018.pdf ](https://drive.google.com/file/d/1b4OaFpgAzYEPo3fKktX3p78pgFHKvAhp/view?usp=sharing)
- [2_eylul_30_11_2018.pdf](https://drive.google.com/file/d/1UVhESfqk4ZSxD7U_brG5Iw4BwFgDkumj/view?usp=sharing)
- [gaziantep_dogus_10_08_2021.pdf](https://drive.google.com/file/d/15O9IbVfXRGAVxnqK00b-uyTVr_Tt89nY/view?usp=sharing)

**Comment by jsvine** [comment]
Hi @yavuzKomecoglu, and thanks for your interest in this library. Your examples and questions are very interesting ones. Unfortunately, I don't have a very conclusive answers. My best guess at what's happening is either:

- The PDF's font definitions are unusual or incorrect. If that's the case, I'm not sure there is much you could do to fix that.

- [`pdfminer.six`](https://github.com/pdfminer/pdfminer.six)'s coordinate-calculation logic is flawed for those fonts. (`pdfminer.six` is the library we use to extract all character coordinates, from which we calculate the word coordinates.) If that's the case, it may be worth filing a discussion or issue on that repository.

My apologies that I can't be more helpful, but feel free to continue the discussion. Perhaps someone else here has deeper insights into what's happening.

**Reply by yavuzKomecoglu** [reply]
Thanks for your feedback.
in this case it only occurs in some header objects in this pdf specific.
As a solution suggestion, I tried to determine the [font metrics values of PIL](https://stackoverflow.com/questions/43060479/how-to-get-the-font-pixel-height-using-pils-imagefont-class) and the blank values from above and below according to different font sizes. This is partially a solution.
